Current master fails compilation: binding for 'error' shadows the existing binding

Weird. Of two platforms, one exhibits this behavior consistently, the other one doesn't. Still, it would be nice to get help determining which one of the installed packages seems to interfere here:
```
cabal: Entering directory '.'
Configuring Agda-2.6.2...
Preprocessing executable 'agda-mode' for Agda-2.6.2..
Building executable 'agda-mode' for Agda-2.6.2..
Preprocessing library for Agda-2.6.2..
Building library for Agda-2.6.2..
[  1 of 371] Compiling Agda.Interaction.ExitCode ( src/full/Agda/Interaction/ExitCode.hs, dist-2.6.2/build/Agda/Interaction/ExitCode.o, dist-2.6.2/build/Agda/Interaction/ExitCode.dyn_o )

src/full/Agda/Interaction/ExitCode.hs:30:34: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘error’ shadows the existing binding
      imported from ‘Prelude’ at src/full/Agda/Interaction/ExitCode.hs:2:8-32
      (and originally defined in ‘GHC.Err’)
   |
30 |                                | error  <- [minBound..maxBound]
   |                                  ^^^^^
cabal: Leaving directory '.'
``` 

MacOS Catalina 10.15.6, Xcode-11.6, Cabal-3.2.0.0, GHC-8.10.1, Agda current master, build with provided `Makefile`.

--- a/Agda.cabal
+++ b/Agda.cabal
@@ -608,42 +608,75 @@ library
   other-modules:    Paths_Agda
 
   -- NOTE: If adding flags, also add to `.ghci-8.0`
-  ghc-options: -fexpose-all-unfoldings
-               -fspecialise-aggressively
-               -fprint-potential-instances
-               -- Initially, we disable all the warnings.
-               -w
-               -Werror
-               -- This option must be the first one after disabling the
-               -- warnings. See Issue #2094.
-               -Wunrecognised-warning-flags
-               -Wdeprecated-flags
-               -Wderiving-typeable
-               -Wdodgy-exports
-               -Wdodgy-foreign-imports
-               -Wdodgy-imports
-               -Wduplicate-exports
-               -Wempty-enumerations
-               -Widentities
-               -Wincomplete-patterns
-               -Winline-rule-shadowing
-               -Wmissing-fields
-               -Wmissing-methods
-               -Wmissing-pattern-synonym-signatures
-               -Wmissing-signatures
-               -Wnoncanonical-monad-instances
-               -Wnoncanonical-monoid-instances
-               -Woverflowed-literals
-               -Woverlapping-patterns
-               -Wsemigroup
-               -Wtabs
-               -Wtyped-holes
-               -Wunrecognised-pragmas
-               -Wunticked-promoted-constructors
-               -Wunused-do-bind
-               -Wunused-foralls
-               -Wwarnings-deprecations
-               -Wwrong-do-bind
+  if impl(ghc <= 8.0.2)
+    ghc-options: -fexpose-all-unfoldings
+                 -fspecialise-aggressively
+                 -fprint-potential-instances
+                 -- Initially, we disable all the warnings.
+                 -w
+                 -Werror
+                 -- This option must be the first one after disabling the
+                 -- warnings. See Issue #2094.
+                 -Wunrecognised-warning-flags
+                 -Wdeprecated-flags
+                 -Wderiving-typeable
+                 -Wdodgy-exports
+                 -Wdodgy-foreign-imports
+                 -Wdodgy-imports
+                 -Wduplicate-exports
+                 -Wempty-enumerations
+                 -Widentities
+                 -Wincomplete-patterns
+                 -Winline-rule-shadowing
+                 -Wmissing-fields
+                 -Wmissing-methods
+                 -Wmissing-pattern-synonym-signatures
+                 -Wmissing-signatures
+                 -Wnoncanonical-monad-instances
+                 -Wnoncanonical-monoid-instances
+                 -Woverflowed-literals
+                 -Woverlapping-patterns
+                 -Wsemigroup
+                 -Wtabs
+                 -Wtyped-holes
+                 -Wunrecognised-pragmas
+                 -Wunticked-promoted-constructors
+                 -Wunused-do-bind
+                 -Wunused-foralls
+                 -Wwarnings-deprecations
+                 -Wwrong-do-bind
+  else
+    ghc-options: -fexpose-all-unfoldings
+                 -fspecialise-aggressively
+                 -fprint-potential-instances
+                 -Werror=unrecognised-warning-flags
+                 -Werror=deprecated-flags
+                 -Werror=deriving-typeable
+                 -Werror=dodgy-exports
+                 -Werror=dodgy-foreign-imports
+                 -Werror=dodgy-imports
+                 -Werror=duplicate-exports
+                 -Werror=empty-enumerations
+                 -Werror=identities
+                 -Werror=incomplete-patterns
+                 -Werror=inline-rule-shadowing
+                 -Werror=missing-fields
+                 -Werror=missing-methods
+                 -Werror=missing-pattern-synonym-signatures
+                 -Werror=missing-signatures
+                 -Werror=noncanonical-monad-instances
+                 -Werror=noncanonical-monoid-instances
+                 -Werror=overflowed-literals
+                 -Werror=overlapping-patterns
+                 -Werror=semigroup
+                 -Werror=tabs
+                 -Werror=typed-holes
+                 -Werror=unrecognised-pragmas
+                 -Werror=unticked-promoted-constructors
+                 -Werror=unused-do-bind
+                 -Werror=unused-foralls
+                 -Werror=warnings-deprecations
+                 -Werror=wrong-do-bind
 
   -- NOTE: If adding flags, also add to `.ghci-8.2`
   if impl(ghc >= 8.2)
@@ -651,14 +684,14 @@ library
                  -- ASR TODO (2017-07-23): `make haddock` fails when
                  -- this flag is on.
                  -- -Wmissing-home-modules
-                 -Wsimplifiable-class-constraints
-                 -Wunbanged-strict-patterns
+                 -Werror=simplifiable-class-constraints
+                 -Werror=unbanged-strict-patterns
 
   -- NOTE: If adding flags, also add to `.ghci-8.6`
   if impl(ghc >= 8.6)
-    ghc-options: -Winaccessible-code
-                 -Wstar-binder
-                 -Wstar-is-type
+    ghc-options: -Werror=inaccessible-code
+                 -Werror=star-binder
+                 -Werror=star-is-type
 
   ghc-prof-options: -fprof-auto
 
@@ -877,53 +910,87 @@ test-suite agda-tests
                      , TypeSynonymInstances
 
   -- NOTE: If adding flags, also add to `.ghci-8.0`
-  ghc-options: -fexpose-all-unfoldings
-               -fspecialise-aggressively
-               -threaded
-               -- Initially, we disable all the warnings.
-               -w
-               -Werror
-               -- This option must be the first one after disabling the
-               -- warnings. See Issue #2094.
-               -Wunrecognised-warning-flags
-               -Wdeprecated-flags
-               -Wderiving-typeable
-               -Wdodgy-exports
-               -Wdodgy-foreign-imports
-               -Wdodgy-imports
-               -Wduplicate-exports
-               -Wempty-enumerations
-               -Widentities
-               -Wincomplete-patterns
-               -Winline-rule-shadowing
-               -Wmissing-fields
-               -Wmissing-methods
-               -Wmissing-pattern-synonym-signatures
-               -Wmissing-signatures
-               -Wnoncanonical-monad-instances
-               -Wnoncanonical-monoid-instances
-               -Woverflowed-literals
-               -Woverlapping-patterns
-               -Wsemigroup
-               -Wtabs
-               -Wtyped-holes
-               -Wunrecognised-pragmas
-               -Wunticked-promoted-constructors
-               -Wunused-do-bind
-               -Wunused-foralls
-               -Wwarnings-deprecations
-               -Wwrong-do-bind
+  if impl(ghc <= 8.0.2)
+    ghc-options: -fexpose-all-unfoldings
+                 -fspecialise-aggressively
+                 -threaded
+                 -- Initially, we disable all the warnings.
+                 -w
+                 -Werror
+                 -- This option must be the first one after disabling the
+                 -- warnings. See Issue #2094.
+                 -Wunrecognised-warning-flags
+                 -Wdeprecated-flags
+                 -Wderiving-typeable
+                 -Wdodgy-exports
+                 -Wdodgy-foreign-imports
+                 -Wdodgy-imports
+                 -Wduplicate-exports
+                 -Wempty-enumerations
+                 -Widentities
+                 -Wincomplete-patterns
+                 -Winline-rule-shadowing
+                 -Wmissing-fields
+                 -Wmissing-methods
+                 -Wmissing-pattern-synonym-signatures
+                 -Wmissing-signatures
+                 -Wnoncanonical-monad-instances
+                 -Wnoncanonical-monoid-instances
+                 -Woverflowed-literals
+                 -Woverlapping-patterns
+                 -Wsemigroup
+                 -Wtabs
+                 -Wtyped-holes
+                 -Wunrecognised-pragmas
+                 -Wunticked-promoted-constructors
+                 -Wunused-do-bind
+                 -Wunused-foralls
+                 -Wwarnings-deprecations
+                 -Wwrong-do-bind
+  else
+    ghc-options: -fexpose-all-unfoldings
+                 -fspecialise-aggressively
+                 -threaded
+                 -Werror=unrecognised-warning-flags
+                 -Werror=deprecated-flags
+                 -Werror=deriving-typeable
+                 -Werror=dodgy-exports
+                 -Werror=dodgy-foreign-imports
+                 -Werror=dodgy-imports
+                 -Werror=duplicate-exports
+                 -Werror=empty-enumerations
+                 -Werror=identities
+                 -Werror=incomplete-patterns
+                 -Werror=inline-rule-shadowing
+                 -Werror=missing-fields
+                 -Werror=missing-methods
+                 -Werror=missing-pattern-synonym-signatures
+                 -Werror=missing-signatures
+                 -Werror=noncanonical-monad-instances
+                 -Werror=noncanonical-monoid-instances
+                 -Werror=overflowed-literals
+                 -Werror=overlapping-patterns
+                 -Werror=semigroup
+                 -Werror=tabs
+                 -Werror=typed-holes
+                 -Werror=unrecognised-pragmas
+                 -Werror=unticked-promoted-constructors
+                 -Werror=unused-do-bind
+                 -Werror=unused-foralls
+                 -Werror=warnings-deprecations
+                 -Werror=wrong-do-bind
 
-  -- ASR (2017-04-11). TODO: Using -Wmissing-home-modules generates an
+  -- ASR (2017-04-11). TODO: Using -Werror=missing-home-modules generates an
   -- error.
   -- NOTE: If adding flags, also add to `.ghci-8.2`
   if impl(ghc >= 8.2)
-    ghc-options: -Wcpp-undef
-                 -Wsimplifiable-class-constraints
-                 -Wunbanged-strict-patterns
+    ghc-options: -Werror=cpp-undef
+                 -Werror=simplifiable-class-constraints
+                 -Werror=unbanged-strict-patterns
 
   -- NOTE: If adding flags, also add to `.ghci-8.6`
   if impl(ghc >= 8.6)
-    ghc-options: -Winaccessible-code
-                 -Wstar-binder
-                 -Wstar-is-type
+    ghc-options: -Werror=inaccessible-code
+                 -Werror=star-binder
+                 -Werror=star-is-type
+

